src/v/storage/disk_log_impl.cc

```diff
@@ -418,11 +419,16 @@ ss::future<compaction_result> disk_log_impl::compact_adjacent_segments(
     std::vector<ss::lw_shared_ptr<segment>> segments;
     std::copy(range.first, range.second, std::back_inserter(segments));
 
-    if (stlog.is_enabled(ss::log_level::debug)) {
-        vlog(stlog.debug, "Compacting {} adjacent segments:", segments.size());
+    if (gclog.is_enabled(ss::log_level::debug)) {
+        std::stringstream segments_str;
         for (size_t i = 0; i < segments.size(); i++) {
-            vlog(stlog.debug, "Segment {}: {}", i + 1, *segments[i]);
+            fmt::print(segments_str, "Segment {}: {}, ", i + 1, *segments[i]);
         }
```

Comment:
This is missing the `Compacting` context now.,

Reply:
it is then appended to the whole log line

Reply:
Sorry, I missed that.